test/instrumentation/modules/aws-sdk/sqs.js

```diff
@@ -0,0 +1,409 @@
+const mockClient = require('../../../_mock_http_client')
+
+const {
+  getToFromFromOperation,
+  getActionFromRequest,
+  getQueueNameFromRequest,
+  getRegionFromRequest,
+  getMessagingDestinationContextFromRequest,
+  shouldIgnoreRequest
+} =
+  require('../../../../lib/instrumentation/modules/aws-sdk/sqs')
+
+initializeAwsSdk()
+
+tape.test('AWS SQS: Unit Test Functions', function (test) {
+  test.test('function getToFromFromOperation', function (t) {
+    t.equals(getToFromFromOperation('deleteMessage'), 'from')
+    t.equals(getToFromFromOperation('deleteMessageBatch'), 'from')
+    t.equals(getToFromFromOperation('receiveMessage'), 'from')
+    t.equals(getToFromFromOperation('sendMessageBatch'), 'to')
+    t.equals(getToFromFromOperation('sendMessage'), 'to')
+    t.end()
+  })
+
+  test.test('function getActionFromOperation', function (t) {
+    const request = {}
+
+    request.operation = 'deleteMessage'
+    t.equals(getActionFromRequest(request), 'delete')
+
+    request.operation = 'deleteMessageBatch'
+    t.equals(getActionFromRequest(request), 'delete_batch')
+
+    request.operation = 'receiveMessage'
+    t.equals(getActionFromRequest(request), 'poll')
+
+    request.operation = 'sendMessage'
+    t.equals(getActionFromRequest(request), 'send')
+
+    request.operation = 'sendMessageBatch'
+    t.equals(getActionFromRequest(request), 'send_batch')
+
+    request.operation = 'sendMessageBatch'
+    request.params = null
+    t.equals(getActionFromRequest(request), 'send_batch')
+
+    request.operation = 'sendMessageBatch'
+    request.params = {}
+    t.equals(getActionFromRequest(request), 'send_batch')
+
+    request.operation = 'receiveMessage'
+    request.params = {}
+    t.equals(getActionFromRequest(request), 'poll')
+
+    request.operation = 'receiveMessage'
+    request.params = { WaitTimeSeconds: 0 }
+    t.equals(getActionFromRequest(request), 'poll')
+
+    request.operation = 'receiveMessage'
+    request.params = { WaitTimeSeconds: -1 }
+    t.equals(getActionFromRequest(request), 'poll')
+
+    request.operation = 'receiveMessage'
+    request.params = { WaitTimeSeconds: 1 }
+    t.equals(getActionFromRequest(request), 'poll')
+    t.end()
+  })
+
+  test.test('function getQueueNameFromRequest', function (t) {
+    const request = {}
+    t.equals(getQueueNameFromRequest(null), 'unknown')
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+
+    request.params = null
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+    request.params = {}
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+
+    request.params.QueueUrl = null
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+    request.params.QueueUrl = 5
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+    request.params.QueueUrl = 'foo/baz/bar'
+    t.equals(getQueueNameFromRequest(request), 'unknown')
+
+    request.params.QueueUrl = 'http://foo/baz/bar'
+    t.equals(getQueueNameFromRequest(request), 'bar')
```

Comment:
Given the "URL" comment above, I would expect this one to fail on node 8.6.

Reply:
Per previous comments -- the `test/instrumentation/modules/aws-sdk/sqs.js` test file wasn't running in CI.  This has been corrected.  We saw Node 8 failures with the above test.  We corrected this by importing the `URL` symbol.